src/main/java/org/esbtools/gateway/resync/controller/ResyncGateway.java

```diff
@@ -0,0 +1,74 @@
+package org.esbtools.gateway.resync.controller;
+
+import org.esbtools.gateway.resync.ResyncRequest;
+import org.esbtools.gateway.resync.ResyncResponse;
+import org.esbtools.gateway.resync.exception.IncompleteRequestException;
+import org.esbtools.gateway.resync.exception.InvalidSystemException;
+import org.esbtools.gateway.resync.exception.ResyncFailedException;
+import org.esbtools.gateway.resync.exception.SystemConfigurationException;
+import org.esbtools.gateway.resync.service.ResyncService;
+import org.esbtools.gateway.resync.service.ResyncServiceRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/")
+public class ResyncGateway {
+
+    protected ResyncServiceRepository resyncServiceRepository;
+
+    @Autowired
+    public ResyncGateway(ResyncServiceRepository resyncServiceRepository) {
+        this.resyncServiceRepository = resyncServiceRepository;
+    }
+
+    @RequestMapping(value="/resync", method=RequestMethod.POST, produces="application/json")
+    public ResponseEntity<ResyncResponse> resync(@RequestBody ResyncRequest resyncRequest) {
+        ResponseEntity<ResyncResponse> responseEntity;
+        try {
+            ResyncService resyncService = resyncServiceRepository.getBySystem(resyncRequest.getSystem());
+            ResyncResponse resyncResponse = resyncService.resync(resyncRequest);
+            responseEntity = new ResponseEntity<ResyncResponse>(resyncResponse, HttpStatus.OK);
+        } catch(InvalidSystemException e) {
+            responseEntity = invalidSystemExceptionHandler(e);
+        } catch (SystemConfigurationException e) {
+            responseEntity = systemConfigurationExceptionHandler(e);
+        } catch(IncompleteRequestException e) {
+            responseEntity = incompleteRequestExceptionHandler(e);
+        } catch(ResyncFailedException e) {
+            responseEntity = resyncFailedExceptionHandler(e);
+        }
```

Comment:
Yes it does actually work running in the container with removing all these try-catch blocks, just wasn't up for changing over all the tests after my last commit last night, so added the try-catch blocks at the very end.  Was going to try and switch to using the MVC unit test framework you mentioned previously today, however I got derailed trying to fold JmsResyncConfiguration into JmsResyncService, and lots of Spring autowiring errors followed.  Will give up on that for now and try to switch get these tests over so we can wrap this PR up.  Will worry about eliminating JmsResyncConfiguration later...